src/pipdeptree/_models/package.py

```diff
@@ -118,21 +110,21 @@ class DistPackage(Package):
 
     """
 
-    def __init__(self, obj: DistInfoDistribution, req: ReqPackage | None = None) -> None:
+    def __init__(self, obj: Distribution, req: ReqPackage | None = None) -> None:
         super().__init__(obj)
         self.req = req
         self._project_name = ""
+        self.editable = False
+        self.direct_url = None
+        self.raw_name = self.project_name
 
     def requires(self) -> list[Requirement]:
-        return self._obj.requires()  # type: ignore[no-untyped-call,no-any-return]
+        return [Requirement(r) for r in self._obj.requires] if self._obj.requires else []
 
     @property
     def project_name(self) -> str:
         if not self._project_name:
-            try:
-                self._project_name = metadata(self.key)["name"]
-            except (PackageNotFoundError, KeyError):
-                self._project_name = self._obj.project_name
+            self._project_name = self.key
         return self._project_name
```

Comment:
We should not assign `self._project_name` to `self.key` since `self.key` is normalized. And let's not do any branching here and instead just grab the name from the `Distribution` object like `Distribution.metadata['Name']`.

Reply:
fixed.